dashboard/app/controllers/test_controller.rb

```diff
@@ -175,4 +184,77 @@ def destroy_level
     level.destroy
     head :ok
   end
+
+  def create_teacher_application
+    return unless (user = current_user)
+    regional_partner = RegionalPartner.create!(name: "regional-partner#{Time.now.to_i}-#{rand(1_000_000)}")
+
+    RegionalPartnerProgramManager.create!(program_manager_id: user.id, regional_partner_id: regional_partner.id)
+
+    teacher_name = "teacher#{Time.now.to_i}-{rand(1_000_000)}"
+    teacher_email = "teacher#{Time.now.to_i}-#{rand(1_000_000)}@test.xx"
+    password = teacher_name + "password"
+    attributes = {
+      name: teacher_name,
+      email: teacher_email,
+      password: password,
+      user_type: "teacher",
+      age: "21+"
+    }
+    teacher = User.create!(attributes)
+
+    form_data = FactoryGirl.build(:pd_teacher_application_hash_common, :csp).to_json
+    application = Pd::Application::TeacherApplication.create!(
+      user: teacher,
+      form_data: form_data,
+      regional_partner_id: regional_partner.id,
+      course: 'csp',
+      status: 'unreviewed'
+    )
+
+    render json: {rp_id: regional_partner.id, teacher_id: teacher.id, application_id: application.id}
+  end
+
+  def create_applications
+    %w(csd csp).each do |course|
```

Comment:
Should we add coverage of csa too? I imagine that doesn't fit within the scope of this PR, just curious if this is something we should keep track of!

Reply:
Hmm, that's a good point, especially since it won't add much time. I need to disable this test soon as my updates to statuses will change the eyes tests, but I'll add an item for this!

Reply:
Thank you!